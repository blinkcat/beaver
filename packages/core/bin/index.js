#!/usr/bin/env node
const { fork } = require('child_process');

const execArgv = [];

if (process.env.NODE_ENABLE_SOURCE_MAP === 'true') {
  execArgv.push('--enable-source-maps');
}

const child = fork(require.resolve('../dist'), process.argv.slice(2), {
  execArgv,
  stdio: 'inherit',
});

process.on('SIGINT', () => {
  child.kill(0);
});
