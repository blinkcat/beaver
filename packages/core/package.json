{
  "name": "@beaver/core",
  "version": "0.1.19",
  "description": "beaver cli core part",
  "main": "./dist",
  "bin": {
    "beaver": "./bin/index.js"
  },
  "scripts": {
    "build": "rm -rf ./dist && tsc",
    "prepublish": "npm run build"
  },
  "files": [
    "bin",
    "dist"
  ],
  "dependencies": {
    "@babel/register": "^7.14.5",
    "@beaver/babel-preset": "^0.1.3",
    "@beaver/types": "^0.1.6",
    "@beaver/utils": "^0.1.3",
    "@loadable/babel-plugin": "^5.13.2",
    "@loadable/server": "5.15.2",
    "@loadable/webpack-plugin": "^5.15.2",
    "crequire": "^1.8.1",
    "escape-string-regexp": "^4.0.0",
    "filesize": "^8.0.7",
    "gzip-size": "^6.0.0",
    "webpack-dev-server": "^4.7.4"
  },
  "devDependencies": {
    "@types/loadable__server": "^5.12.6",
    "typescript": "^4.3.4",
    "webpack": "^5.45.1"
  },
  "peerDependencies": {
    "webpack": "^4.37.0 || ^5.0.0"
  },
  "keywords": [],
  "author": "",
  "gitHead": "0546b4b996e0023f18465d22b7f8bde958643be1"
}
