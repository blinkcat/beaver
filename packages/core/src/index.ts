import { mri } from '@beaver/utils';
import pluginManager from './pluginManager';
import configManager from './configManager';
import configPlugin, { hooks as configPluginHooks, resolvedConfigProxy } from './plugin/config';
import pathsPlugin, { hooks as pathsPluginHooks } from './plugin/paths';
import htmlPlugin from './plugin/html';
import ssgPlugin from './plugin/ssg';
import devCommandPlugin from './plugin/command/dev';
import buildCommandPlugin from './plugin/command/build';
import paths from './paths';

export default async function init() {
  // load config first, so we can get user's plugins
  configManager.loadConfig();
  // then, prepend internal plugins, and load all these plugins
  pluginManager.loadPlugins([
    configPlugin,
    pathsPlugin,
    htmlPlugin,
    ssgPlugin,
    devCommandPlugin,
    buildCommandPlugin,
    ...(configManager.inputConfig?.plugins || []),
  ]);

  // trigger internal hooks
  configManager.resolveConfig(await configPluginHooks.config.promise({ ...configManager.inputConfig }));
  Object.assign(paths, await pathsPluginHooks.paths.promise(paths, { config: resolvedConfigProxy }));

  const {
    _: [command],
    ...args
  } = mri(process.argv.slice(2));

  pluginManager.receiveCommand(command, args);
}

init();
