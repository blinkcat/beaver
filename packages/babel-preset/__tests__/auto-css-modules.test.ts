import { transformSync } from '@babel/core';
import babelPluginAutoCssModules from '../babel-plugins/auto-css-modules';

function transform(code: string) {
  return transformSync(code, {
    filename: 'test.js',
    plugins: [babelPluginAutoCssModules],
  });
}

['css', 'less', 'scss', 'sass'].forEach(ext => {
  test(`works on ${ext} file`, () => {
    const ret = transform(`import styles from './test.${ext}'`);
    expect(ret?.code).toMatchSnapshot();
  });
});

test('ignores directly import', () => {
  const ret = transform(`import './test.css'`);
  expect(ret?.code).toMatchSnapshot();
});
