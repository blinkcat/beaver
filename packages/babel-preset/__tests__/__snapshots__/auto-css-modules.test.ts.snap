// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignores directly import 1`] = `
"\\"use strict\\";

require(\\"./test.css\\");"
`;

exports[`works on css file 1`] = `
"\\"use strict\\";

var _test = _interopRequireDefault(require(\\"./test.css?modules\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }"
`;

exports[`works on less file 1`] = `
"\\"use strict\\";

var _test = _interopRequireDefault(require(\\"./test.less?modules\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }"
`;

exports[`works on sass file 1`] = `
"\\"use strict\\";

var _test = _interopRequireDefault(require(\\"./test.sass?modules\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }"
`;

exports[`works on scss file 1`] = `
"\\"use strict\\";

var _test = _interopRequireDefault(require(\\"./test.scss?modules\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }"
`;
